diff --git a/kernel/sched/fair.c b/kernel/sched/fair.c
index 9618332d80e1..5c6330bad2f0 100644
--- a/kernel/sched/fair.c
+++ b/kernel/sched/fair.c
@@ -5008,6 +5008,7 @@ enqueue_task_fair(struct rq *rq, struct task_struct *p, int flags)
 #ifdef CONFIG_SMP
 	int task_new = flags & ENQUEUE_WAKEUP_NEW;
 #endif
+	bool prefer_idle = schedtune_prefer_idle(p) > 0;
 
 	/*
 	 * The code below (indirectly) updates schedutil which looks at
@@ -5022,7 +5023,7 @@ enqueue_task_fair(struct rq *rq, struct task_struct *p, int flags)
 	 * utilization updates, so do it here explicitly with the IOWAIT flag
 	 * passed.
 	 */
-	if (p->in_iowait)
+	if (p->in_iowait && prefer_idle)
 		cpufreq_update_util(rq, SCHED_CPUFREQ_IOWAIT);
 
 	for_each_sched_entity(se) {
